data_prep/reuters_data.py

```diff
@@ -0,0 +1,133 @@
+        """
+        Download the dataset and filters out all documents which have more or less than 1 class.
+        Then filters out all classes which have no remaining documents.
+        Args:
+            r8 (bool, optional): R8 is constructed by taking only the top 10 (original) classes. Defaults to False.
+            val_size (float, optional): Proportion of training documents to include in the validation set.
+        Returns:
+            doc_splits (tuple): Tuple containing 3 List of training, test, and validation documents.
+            unique_classes (List): List of Strings containing the class names sorted in alphabetical order.
+        """
+        nltk.download('reuters')
+        # Filter out docs which don't have exactly 1 class
+        data = defaultdict(lambda: {'train': [], 'test': []})
+
+        for doc in reuters.fileids():
+            categories = reuters.categories(doc)
+            if len(categories) == 1:
+                if doc.startswith('training'):
+                    data[categories[0]]['train'].append(doc)
+                if doc.startswith('test'):
+                    data[categories[0]]['test'].append(doc)
+
+        # Filter out classes which have no remaining docs
+        for cat in reuters.categories():
+            if len(data[cat]['train']) < 1 or len(data[cat]['test']) < 1:
+                data.pop(cat, None)
+
+        if r8:
+            # Choose top 10 classes and then select the ones which still remain after filtering
+            popular = sorted(reuters.categories(), key=lambda clz: len(reuters.fileids(clz)), reverse=True)[:10]
+            data = dict([(cls, splits) for (cls, splits) in data.items() if cls in popular])
+
+        # Create splits
+        train_docs = [doc for cls, splits in data.items() for doc in splits['train']]
+        test_docs = [doc for cls, splits in data.items() for doc in splits['test']]
+
+        # Select the validation documents out of the training documents
+        val_size = int(len(train_docs) * val_size)
+        random.shuffle(train_docs)
+        val_docs = train_docs[:val_size]
+        train_docs = train_docs[val_size:]
+
+        # sort the unique classes to ensure constant order
+        unique_classes = sorted(data.keys())
+        return (train_docs, test_docs, val_docs), unique_classes
+    
+    @staticmethod
+    def _prepare_split(docs, classes):
+        """
+        Extract the text and labels of each documents.
+        Args:
+            docs (List): List of document names from which the texts and labels will be extracted.
+            classes (List): List of unique class names sorted in alphabetical order.
+        Returns:
+            texts (List): List of Strings containing the text of the documents.
+            labels (List): List of int containing the labels of the documents.
+        """
+        texts = []
+        labels = []
+        for doc in docs:
+            text = ' '.join(reuters.words(doc))
```

Comment:
use `reuters.raw(doc)` instead as "Matthias." and "Matthias ." is different according to Roberta (and it's also much nicer)